### ISSUE TITLE
QOL improvement: allow integer values for `Optimizer.set_learning_rate()`

### ISSUE THREAD

**Issue by Speierers** [mitsuba-renderer/mitsuba3#264]
### Discussed in https://github.com/mitsuba-renderer/mitsuba3/discussions/257

<div type='discussions-op-text'>

<sup>Originally posted by **maxfrei750** September 16, 2022</sup>
Sometimes, I forget to specify the learning rate with a decimal point and get
```
Exception has occurred: Exception
Optimizer.set_learning_rate(): value should be a float or a dict!
  File "/src/mitsuba3/build/python/mitsuba/python/ad/optimizers.py", line 103, in set_learning_rate
    raise Exception('Optimizer.set_learning_rate(): value should be a float or a dict!')
  File "/src/mitsuba3/build/python/mitsuba/python/ad/optimizers.py", line 21, in __init__
    self.set_learning_rate(lr)
  File "/src/mitsuba3/build/python/mitsuba/python/ad/optimizers.py", line 261, in __init__
    super().__init__(lr, params)
```

Would it result in problems, if we added

```
if isinstance(lr, int):
    lr = float(lr)
```
to https://github.com/mitsuba-renderer/mitsuba3/blob/6902109b9cab4607eaab47fa00bdd196ee458e51/src/python/python/ad/optimizers.py#L83-L103
?</div>

**Comment by Speierers** [comment]
This was fixed in [53143db](https://github.com/mitsuba-renderer/mitsuba3/commit/53143db05739b964b7a489f58dbd1bd4da87533c)
